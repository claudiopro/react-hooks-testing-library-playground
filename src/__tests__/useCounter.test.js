import { renderHook, act } from "@testing-library/react-hooks";
import useCounter from "../useCounter";

test("should use counter", () => {
  const { result } = renderHook(() => useCounter());

  expect(result.current.count).toBe(0);
  expect(typeof result.current.increment).toBe("function");
});

test("should increment counter", () => {
  const { result } = renderHook(() => useCounter());

  act(() => {
    result.current.increment();
  });

  expect(result.current.count).toBe(1);
});

test("should match snapshot", () => {
  const { result } = renderHook(() => useCounter());

  expect(result).toMatchSnapshot();

  act(() => {
    result.current.increment();
  });

  expect(result).toMatchSnapshot();
});
