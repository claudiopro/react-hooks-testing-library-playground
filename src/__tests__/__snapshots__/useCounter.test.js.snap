// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
Object {
  "all": Array [
    Object {
      "count": 0,
      "increment": [Function],
    },
  ],
  "current": Object {
    "count": 0,
    "increment": [Function],
  },
  "error": undefined,
}
`;

exports[`should match snapshot 2`] = `
Object {
  "all": Array [
    Object {
      "count": 0,
      "increment": [Function],
    },
    Object {
      "count": 1,
      "increment": [Function],
    },
  ],
  "current": Object {
    "count": 1,
    "increment": [Function],
  },
  "error": undefined,
}
`;
